packages/edit-site/src/components/block-editor/get-block-editor-provider.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import DefaultBlockEditor from './providers/default-block-editor-provider';
+import NavigationBlockEditor from './providers/navigation-block-editor-provider';
+
+/**
+ * Factory to isolate choosing the appropriate block editor
+ * component to handle a given entity type.
+ *
+ * @param {string} entityType the entity type being edited
+ * @return {JSX.Element} the block editor component to use.
+ */
+export default function getBlockEditorProvider( entityType ) {
+	let Provider = null;
+
+	switch ( entityType ) {
+		case 'wp_navigation':
+			Provider = NavigationBlockEditor;
+			break;
+		case 'wp_template':
+		case 'wp_template_part':
+		default:
+			Provider = DefaultBlockEditor;
+			break;
+	}
+
+	return Provider;
+}
```

Comment:
It feels like this file could be avoided entirely by just doing this one liner inline:

```
const Provider = entityType === 'wp_navigation' ? NavigationBlockEditor : DefaultBlockEditor;
```

I realize that we could have potentially more later, but maybe that ternary is sufficient for now.

Reply:
The motivation is not to over-engineer the solution and remove potentially unnecessary indirection in the code - am I right?

I'm not hugely set on keeping this, but my only counter would be that the dedicated function clearly demarcates which entities are served by which provider whereas the ternary only tells us about the Navigation entities.

That said the ternary does express the difference clearly, indicating which of the entities is the edge case so maybe that's ok.

Reply:
I'm fine either way 👍